### ISSUE TITLE
Reproduce the results of Insertion Transformer on WMT17 En-De

### ISSUE THREAD

**Issue by Varal7** [pytorch/fairseq#1428]
Hi,

I've tried reproducing the results of the [Insertion Transformation](https://arxiv.org/abs/1902.03249) on En-De translation tasks.

Following the readme from this repository, I've used the training scripts with the following arguments:

```
bash prepare-wmt14en2de.sh

cd ../..

TEXT=examples/translation/wmt17_en_de
fairseq-preprocess \
    --source-lang en --target-lang de \
    --trainpref $TEXT/train --validpref $TEXT/valid --testpref $TEXT/test \
    --destdir data-bin/wmt17_en_de --thresholdtgt 0 --thresholdsrc 0 \
    --workers 20 \
    --joined-dictionary
```

```
fairseq-train \
    data-bin/wmt17_en_de \
    --save-dir checkpoints \
    --ddp-backend=no_c10d \
    --task translation_lev \
    --criterion nat_loss \
    --arch insertion_transformer \
    --noise random_delete \
    --share-all-embeddings \
    --optimizer adam --adam-betas '(0.9,0.98)' \
    --lr 0.0005 --lr-scheduler inverse_sqrt \
    --min-lr '1e-09' --warmup-updates 10000 \
    --warmup-init-lr '1e-07' --label-smoothing 0.1 \
    --dropout 0.3 --weight-decay 0.01 \
    --decoder-learned-pos \
    --encoder-learned-pos \
    --apply-bert-init \
    --log-format 'simple' --log-interval 100 \
    --fixed-validation-seed 7 \
    --max-tokens 4000 \
    --save-interval-updates 10000 \
    --max-update 300000 \
    --fp16 \
    --memory-efficient-fp16 \
    --tensorboard-logdir /data/rsg/nlp/quach/serialize/insertion_transformer/wmt17
```

```
fairseq-generate \
  data-bin/wmt17_en_de \
 --gen-subset test \
 --task translation_lev \
 --path checkpoints/checkpoint_best.pt \
 --iter-decode-max-iter 9 \
 --iter-decode-eos-penalty 2 \
 --beam 5 --remove-bpe \
 --print-step \
 --batch-size 200
```

This achieves a BLEU score of 18.55. The paper reports BLEU scores in the high 20s (26.72 when using uniform training)

Now I know that this is the WMT17 task instead of the WMT14 and that I'm not using distillation, but 18.55 seems very low. Am I missing something?

Thanks!

**Comment by huihuifan** [comment]
@kahne ?

**Comment by MultiPath** [comment]
Hi, based on my own experience, if you trained your model on real data, it will give you very low scores, especially using "uniform slot loss".

**Comment by alphadl** [comment]
@Varal7 Have you reproduced the result? I met the same question with you, where I trained on the distilled data and got only 21.61 BLEU
